lib/core/src/Cardano/Wallet.hs

```diff
@@ -249,20 +284,28 @@ mkWalletLayer db nw tl = WalletLayer
 
     { createWallet = \wid wname s -> do
         let checkpoint = initWallet s
-        now <- liftIO getCurrentTime
         let metadata = WalletMetadata
                 { name = wname
-                , passphraseInfo = WalletPassphraseInfo now
+                , passphraseInfo = Nothing
                 , status = Restoring minBound
                 , delegation = NotDelegating
                 }
         DB.createWallet db (PrimaryKey wid) checkpoint metadata $> wid
 
     , readWallet = _readWallet
 
+    , updateWallet = \wid modify -> DB.withLock db $ do
+        meta <- _readWalletMeta wid
+        DB.putWalletMeta db (PrimaryKey wid) (modify meta)
+
+    , updateWalletPassphrase = \wid (old, new) -> do
```

Comment:
I wonder what happens if there are two requests at the same time in a rece condition event both calling `updateWalletPassphrase`

Reply:
There's a DB lock, so ultimately, they'll be sequentialized.

Reply:
https://github.com/input-output-hk/cardano-wallet/pull/249/files#r282877383